src/use_case/login/LoginInteractor.java

```diff
@@ -0,0 +1,56 @@
+package use_case.login;
+
+import java.util.ArrayList;
+/*
+Missing Project entity
+ */
+
+public class LoginInteractor implements LoginInputBoundary {
+    private final LoginDataAccessInterface userDataAccessObject;
+    private final LoginOutputBoundary loginPresenter;
+    private ArrayList<ArrayList<String>> projectData;
+
+    public LoginInteractor(LoginDataAccessInterface userDataAccessInterface,
+                           LoginOutputBoundary loginOutputBoundary) {
+        this.userDataAccessObject = userDataAccessInterface;
+        this.loginPresenter = loginOutputBoundary;
+    }
+
+    @Override
+    public void execute(LoginInputData loginInputData) {
+        String username = loginInputData.getUsername();
+        String password = loginInputData.getPassword();
+        if (userDataAccessObject.getUser(username) == null) {
+            loginPresenter.prepareFailView(username + ": Account does not exist.");
+        } else {
+            String pwd = userDataAccessObject.getUser(username).getPassword(); // User needs a method getPassword()
+            if (!password.equals(pwd)) {
+                loginPresenter.prepareFailView("Incorrect password for " + username + ".");
+            } else {
+
+                User user = userDataAccessObject.getUser(loginInputData.getUsername());
```

Comment:
Not a big deal but loginInputData.getUsername() is already saved in the variable username, so there isn't a need to call this method again.